### PR TITLE
[py]: Propagate `stderr` to exceptions when selenium manager fails

### PR DESCRIPTION
**Pull request by symonk**
Capture `stderr` when calling selenium manager and propogate the error to the user.  This propagates `stderr` to the user (and `stdout` for now).  When selenium manager exits `0` we pass on the stripped path returned from its stdout.

adds some unit tests for the manager aswell as we currently have none.  I've made the `SeleniunManager` class require an instance to use as its a single place we use it and I think we may want to build some internal state soon, rather than have a bunch of static methods in a class calling each other from the class directly.

Thanks!

Relates to #11301

### CODE REVIEW

**Comment by symonk** [comment]
Did CI change? I'd definitely expect full python tests to run against this diff?

File: py/selenium/webdriver/common/service.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def start(self) -> None:
                 logger.debug("driver not found in PATH, trying Selenium Manager")
                 browser = self.__class__.__module__.split(".")[-2]
                 try:
-                    path = SeleniumManager.driver_location(browser)
+                    path = SeleniumManager().driver_location(browser)
                 except WebDriverException as new_err:
                     logger.debug("Unable to obtain driver using Selenium Manager: " + new_err.msg)
```
Comment (symonk) [inline py/selenium/webdriver/common/service.py:98]: I _think_ this is sufficient to propagate here from the changes, should include the message tho unit tests do not currently test an actual `service` based flow; only exercising the manager directly at the moment.

**Comment by symonk** [comment]
`selenium-manager` binary is not `.gitignored` in `/py/*` so i've accidentally committed here, will fix this PR when I have free time after work

**Comment by titusfortner** [comment]
Hah, yes, we need to add them to .gitignore. Sorry & thanks.

File: py/selenium/webdriver/common/selenium_manager.py [hunk 2]
```diff
@@ -79,9 +82,10 @@ def run(args: Tuple[str, str, str]) -> str:
         :Returns: The log string containing the driver location.
         """
         logger.debug(f"Executing selenium manager with: {args}")
-        result = subprocess.run(args, stdout=subprocess.PIPE).stdout.decode("utf-8")
-
-        if not re.match("^INFO\t", result):
-            raise WebDriverException(f"Unsuccessful command executed: {args}")
-
-        return result
+        completed_proc = subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
Comment (titusfortner) [inline py/selenium/webdriver/common/selenium_manager.py:86]: I think we need to also rescue this with something like: 
```
raise SeleniumManagerException(f"Unsuccessful command executed. {args}; {error}")
```

File: py/selenium/webdriver/common/selenium_manager.py [hunk 3]
```diff
@@ -79,9 +82,10 @@ def run(args: Tuple[str, str, str]) -> str:
...
+        stdout, stderr = completed_proc.stdout.decode("utf-8"), completed_proc.stderr.decode("utf-8")
+        if completed_proc.returncode:
+            raise SeleniumManagerException(f"Selenium Manager exited non zero.  {stdout}{stderr}")
```
Comment (titusfortner) [inline py/selenium/webdriver/common/selenium_manager.py]: want to keep what command was executed in the error message...

File: py/selenium/webdriver/common/service.py (hunk 1 above)
Comment (titusfortner) [reply py/selenium/webdriver/common/service.py:98]: yeah, we're going to move to instantiating this class for next set of features, I think.

**Review by titusfortner — CHANGES_REQUESTED** [review]

**Comment by symonk** [comment]
important to squash here

**Pull request merged**